### PR TITLE
fix: check versioned hash version for EIP-4844 transactions

### PR DESCRIPTION
**Pull request by jtraglia**
I didn't see where the versioned hash's version was checked for EIP-4844 transactions. This PR adds this check. I don't believe this is a big deal though because the CL should also check this.

Here's the spec for this:
* https://eips.ethereum.org/EIPS/eip-4844#execution-layer-validation

This part in particular:
```python
# all versioned blob hashes must start with VERSIONED_HASH_VERSION_KZG
for h in tx.blob_versioned_hashes:
    assert h[0] == VERSIONED_HASH_VERSION_KZG
```

### CODE REVIEW

**Comment by Rjected** [comment]
it should be checked here, although implicitly by recalculating the versioned hash
https://github.com/paradigmxyz/reth/blob/1cc52ff73cbe268e863434ee3eb8c646787831bf/crates/primitives/src/transaction/eip4844.rs#L147-L150

the error type is useful though, see the TODO in the above snippet

**Comment by jtraglia** [comment]
Oh I see, thanks! Would it be worth keeping both checks? How should I proceed?

You're welcome to modify this PR how you see fit btw.

File: crates/transaction-pool/src/validate/eth.rs [hunk 1]
```diff
@@ -285,6 +286,24 @@ where
...
+                            InvalidPoolTransactionError::Eip4844(
+                                Eip4844PoolTransactionError::WrongEip4844HashVersion {
+                                    have: version,
+                                    permitted: VERSIONED_HASH_VERSION_KZG,
+                                },
+                            ),
+                        );
+                    }
+                }
+            }
```
Comment (Rjected) [inline crates/transaction-pool/src/validate/eth.rs:306]: would remove this

File: crates/transaction-pool/src/error.rs [hunk 2]
```diff
@@ -143,17 +143,25 @@ pub enum Eip4844PoolTransactionError {
...
         permitted: usize,
     },
+    /// Thrown if an EIP-4844 transaction with the wrong versioned hash version arrives
+    #[error("wrong versioned hash version: have {have}, permitted {permitted}")]
+    WrongEip4844HashVersion {
+        /// The transaction's versioned hash version
+        have: u8,
+        /// The EIP-4844 versioned hash version
+        permitted: u8,
+    },
```
Comment (Rjected) [inline crates/transaction-pool/src/error.rs:164]: would create a new variant here instead
https://github.com/paradigmxyz/reth/blob/1cc52ff73cbe268e863434ee3eb8c646787831bf/crates/primitives/src/transaction/sidecar.rs#L30

**Review by Rjected — CHANGES_REQUESTED** [review]
I think the way forward would be to create a new `BlobTransactionValidationError` variant that describes a mismatched versioned hash, and change this to return that variant
https://github.com/paradigmxyz/reth/blob/1cc52ff73cbe268e863434ee3eb8c646787831bf/crates/primitives/src/transaction/eip4844.rs#L149

**Comment by Rjected** [comment]
adding a link to the spec, or even quoting the spec code verbatim around those lines would be useful as an artifact of this

**Comment by jtraglia** [comment]
Sounds good. I'll make this change tomorrow.

**Comment by jtraglia** [comment]
Should be ready now!

**Review by Rjected — APPROVED** [review]
awesome, this looks good to me, thanks!

**Pull request merged**